### DISCUSSION TITLE
HTML site maps and their role in accessibility

### DISCUSSION THREAD

**Discussion by amycole501** [uswds/uswds#5561] [Accessibility]
Does your website have an HTML site map? If not, consider adding one. HTML site maps are [recommended by WCAG. ](https://www.w3.org/TR/WCAG20-TECHS/G63.html)They allow users to navigate the site outside of top-level navigation, which conforms to WCAG [Success Criterion 2.4.5, Multiple Ways](https://www.w3.org/TR/2008/REC-WCAG20-20081211/#navigation-mechanisms-mult-loc).

HTML site maps become especially important for visitors using assistive technologies who may have difficulty navigating large dropdown menus. Site maps also provide an “at-a-glance” list of all pages on a site and their relationship to one another.

Site maps, like many accessibility accommodations, boost search engine optimization (SEO) — they  allow Google and other search engines to easily index pages not otherwise linked across your site.

USWDS’s own website isn’t using a site map right now, but we intend to add one in the future. Please share your experience or questions about HTML site maps below. We hope that your insights will inform our own progress towards an HTML site map.

**Comment by bonnieAcameron** [comment]
I'm not even sure how to go about creating and implementing a site map... If anyone has any preferred resources on how to build one so that I can get an idea of scope, expertise needed, and steps, I'd love to look through them.

**Reply by jcaseywhite** [reply]
The scope may depend on how large the site is, and if the site is static or on a CMS such as Drupal. With larger sites, it will be beneficial to create a sitemap index that references other sitemaps, in order to prevent cluttering. The expertise to create an HTML sitemap is traditionally planned by someone with UX and/or SEO knowledge. However, based on recent experience with Accessibility, they should also be included in that planning process. If you want links on a sitemap to automatically update, as content adds more pages to the website, you will need Dev involved. Also, if the website is static, without a CMS, Dev will play a larger role in creating an HTML sitemap.

To find examples try typing popular sites with the word "sitemap" into Google. It should feed you HTML Sitemap results. You will see how different companies make sitemaps and get ideas. However, your sitemap will be unique based on how your site is currently structured. The industry you serve could also affect how your sitemap is organized. When researching you find that large news organizations such as CNN create a main Sitemap Index by year, and that their subordinate sitemaps separate further into major topics. International websites might decide to organize sitemaps into languages instead.

Currently, VOTE has very simple sitemaps for each language. Their sitemaps can be improved upon, but they are a good example for sites with less page content. https://vote.gov/sitemap/

**Reply by amycole501** [reply]
Some sites are enormous and are added to via a CMS, so you may want to structure the HTML site map based on primary pages you want to index and as Casey mentioned, involve the dev team to help automatically add to a site map if possible. Also important are in-page navigation links such as breadcrumbs and "related link" features. Bottom line, if a human can click on a link with a keyboard or mouse, search engines can also crawl/follow link paths. Most people land on a site's interior pages based on specific keyword queries, so providing the HTML site map in the footer and breadcrumb navigation helps orient a person more easily, especially visitors using assistive technology.

**Comment by jcaseywhite** [comment]
As an SEO, I want to reiterate what @amycole501 said about navigating past dropdowns. I have found that sitemaps are incredibly powerful for crawlers to bypass pages that have heavy JavaScript navigation features that require query strings. This is especially true on larger sites that use faceted navigation (also known as 'faceted search' or 'filters').  An example of a URL with a query string:

https://example.com/page?field1=random&field2=filter

Many government sites have pages that remain hidden due to this, which prevents them from being indexed properly by search engines such as Google. Accessibility plays a big role in HTML sitemaps because the HTML version of sitemaps can assist those with visual impairments who have issues navigating custom page filter tools on a page.

There are multiple ways to organize HTML sitemaps. What I personally like to suggest is a main Sitemap Index that links to subordinate sitemaps that further organize articles by the year they were last modified or separate language content.

However, if you want sitemaps that provide more specific information to search engine crawlers and that can be uploaded into Google Search Console (GSC) you will want to look into XML sitemaps.

**Reply by amycole501** [reply]
The combo approach of XML and HTML site maps is a great recipe for success. And excellent point about multiple languages! More sites are offering human-translated HTML pages so additional navigation options are key to proper indexing.

**Comment by mgifford** [comment]
From an accessibility point of view, governments should be making sure that their content is as accessible as possible. Sitemaps & XML Sitemaps are great as they provide an alternative way for people to see content on a site. XML Sitemaps usually just affect how people see the content in their favorite search engine. Micro-content like RDFA that allows other tools to pull meaning from your pages is even better https://developers.google.com/custom-search/docs/structured_data

If a citizen can find a trustworthy answer to a question they have without ever accessing a government website, that sounds like a win for everyone.
